plugins/modules/entrust_cagw_certificate.py

```diff
@@ -0,0 +1,860 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c), Entrust Datacard Corporation, 2019
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: entrust_cagw_certificate
+author:
+    - Sapna Jain (@jains)
+short_description: Request SSL/TLS certificates with the Certificate Authority Gateway (CAGW) API
+description:
+    - Create, get, and take actions (Hold, Unhold, Revoke certificates) with the Certificate Authority Gateway (CAGW) API.
+    - Requires credentials for calling the CAGW API.
+notes:
+    - C(path) must be specified as the output location of the certificate.
+requirements:
+    - cryptography >= 1.6
+options:
+    force:
+        description:
+            - If force is used, a certificate is requested regardless of whether I(path) points to an existing valid certificate.
+        type: bool
+        default: False
+    path:
+        description:
+            - The destination path for the generated certificate as a PEM encoded cert.
+            - If there is already an Entrust certificate at this location, it will be replaced always.
+            - If enrollment_format is PKCS12 then it will have Base64 encoded PKCS12 body
+        type: path
+    csr:
+        description:
+            - Base-64 encoded Certificate Signing Request (CSR). csr is accepted without PEM formatting around the Base-64 string.
+            - If no csr is provided when request_type=new and enrollment_format=X509, the certificate will not be generated and module will be failed
+        type: path
+
+    cagw_api_client_cert_path:
+        description:
+            - Path for the Client cert issued by the same CA
+        type: path
+        required: true
+
+    cagw_api_client_cert_key_path:
+        description:
+            - Path for the Client cert key issued by the same CA
+        type: path
+        required: true
+
+    host:
+        description:
+            - Host or ip address for Entrust CAGW
+        type: str
+        required: true
+
+    port:
+        description:
+            - port for Entrust CAGW
+        type: str
+        default: 443
+
+    certificate_authority_id:
+        description:
+            - Unique id for the Certificate Authority
+        type: str
+        required: true
+
+    certificate_profile_id:
+        description:
+            - Profile id for the Certificate Authority
+        type: str
+
+    request_type:
+        description:
+            - request type i.e. new (stands for enrollment), get (stands for get certificate), action (stands for action to be taken on the certificate)
+        type: str
+        choices: [ 'new', 'action', 'get' ]
+        required: true
```

Comment:
The UX might be cleaner if you have multiple modules for these operations. Like `entrust_cagw_certificate_info` for getting a certificate, `entrust_cagw_certificate` for creating a new one, and `entrust_cagw_certificate_action` for applying an action.

Reply:
I have done this mapping vis-a-vis the certificate module within the Entrust CA Gateway where the operations i.e. new, action, and get are like crud operations on an entity i.e. certificate. I can break the module but that will be slightly mis-aligned with the CAGW API specs. Do you see this as a must have?